.travis.yml

```diff
@@ -33,15 +33,22 @@ matrix:
 #            - BREW_INSTALL=python3
 
 install:
-  - pip install tox
+  - pip install flake8 tox
 #  - |
 #    if [[ $TRAVIS_OS_NAME == 'osx' ]]; then
 #      if [[ -n "$BREW_INSTALL" ]]; then
 #        brew update
 #        brew install "$BREW_INSTALL"
 #      fi
 #    fi
-#    pip install tox
+
+before_script:
+  # stop the build if there are Python syntax errors or undefined names
+  # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
+  if [[ $TOXENV == py37 ]]; then
+    flake8 . --count --exclude=./.,./examples,./tests --select=E901,E999,F821,F822,F823 --show-source --statistics ;
```

Comment:
I'm not sure how the exclude parser for ``flake8`` works, but the way you have it now actually excludes **EVERYTHING**.  The pattern needs to be written with a trailing asterisk "./.*" like so:
```sh
flake8 . --count --exclude=./.*,./examples,./tests --select=E901,E999,F821,F822,F823 --show-source --statistics ;
```

Same comment applies to the line below.

Reply:
I agree.

Undoing the last change.

Reply:
@cclauss The last change was a good change, it just needed an asterisk after the dot in the first part of the excludes.

Without that exclude it runs for an extremely long time on my system - many minutes.

Reply:
???

$ __time flake8 .__
```
real	0m3.379s
user	0m7.075s
sys	0m0.268s
```
$ __time flake8 . --exclude=./.*__
```
real	0m3.161s
user	0m7.070s
sys	0m0.224s
```
What is your machine?

Reply:
Try a **time flake8 --exclude=./examples,./tests**.

The timing is fine if you don't have any excludes - I'm assuming it has sane default exclusions if you don't specify --exclude.  But as soon as you specify --exclude, you need to do a thorough job of it - it looks like those sane defaults go away.